pkg/resources/cruisecontrol/cruisecontrol.go

```diff
@@ -148,25 +154,39 @@ func (r *Reconciler) Reconcile(log logr.Logger) error {
 }
 
 func (r *Reconciler) getClientPassword() (string, error) {
-	var clientPass string
-	if r.KafkaCluster.Spec.IsClientSSLSecretPresent() {
-		// Use that secret as default which has autogenerated for clients by us
-		clientNamespacedName := types.NamespacedName{Name: fmt.Sprintf(pkicommon.BrokerControllerTemplate, r.KafkaCluster.Name), Namespace: r.KafkaCluster.Namespace}
-		if r.KafkaCluster.Spec.GetClientSSLCertSecretName() != "" {
-			clientNamespacedName = types.NamespacedName{Name: r.KafkaCluster.Spec.GetClientSSLCertSecretName(), Namespace: r.KafkaCluster.Namespace}
+	clientSecret, err := r.getClientSecret()
+	if err != nil {
+		return "", err
+	}
+	return string(clientSecret.Data[v1alpha1.PasswordKey]), nil
+}
+
+func (r *Reconciler) getClientSecret() (*corev1.Secret, error) {
+	clientSecret := &corev1.Secret{}
+	// Use that secret as default which has autogenerated for clients by us
+	clientNamespacedName := types.NamespacedName{Name: fmt.Sprintf(pkicommon.BrokerControllerTemplate, r.KafkaCluster.Name), Namespace: r.KafkaCluster.Namespace}
+	if r.KafkaCluster.Spec.GetClientSSLCertSecretName() != "" {
+		clientNamespacedName = types.NamespacedName{Name: r.KafkaCluster.Spec.GetClientSSLCertSecretName(), Namespace: r.KafkaCluster.Namespace}
+	}
+
+	if err := r.Client.Get(context.TODO(), clientNamespacedName, clientSecret); err != nil {
```

Comment:
**Recommended**: We **SHOULD** prefer using [`context.Background()`](https://pkg.go.dev/context#Background) to [`context.TODO()`](https://pkg.go.dev/context#TODO) unless we are planning to refactor the function to pass down a context here, because otherwise this is a top level request context in this case.